db/seeds/api.controller.js

```diff
@@ -1,8 +1,53 @@
 const endpointsJson = require("../../endpoints.json");
+const articles = require("../data/test-data/articles");
+const { checkArticleExists } = require("./api.articles.model");
+const { selectTopics, selectArticle } = require("./api.model");
 
-function getApi(req, res) {
+console.log("in controller");
+
+exports.getApi = (req, res) => {
   console.log(endpointsJson, "<--- endpoints json");
   res.status(200).send({ endpoint: endpointsJson });
-}
+};
+
+exports.getTopics = (req, res, next) => {
+  selectTopics()
+    .then((topics) => {
+      if (topics.length === 0 || !topics) {
+        const error = new Error("Not Found");
+        error.status = 404;
+        return next(error);
+      }
+      res.status(200).send({ topics });
+    })
+    .catch(next);
+};
+
+exports.getArticle = (req, res, next) => {
+  const { article_id, sort_by } = req.query;
+  const promises = [selectArticle(article_id, sort_by)];
+  if (article_id) {
+    promises.push(checkArticleExists(article_id));
+  }
```

Comment:
You don't really need this logic for this type of endpoint because articles either exists, or they don't in which case your error handling should deal with it, this would be reserved for when you get to things like "comments" because the article might not exist in which case you get and error OR the article DOES exist but there are no comments on it, in which case you should get an empty array.